File: src/test/java/model/TreeParser.java
package model;

import com.sun.source.tree.Tree;
import org.junit.jupiter.api.Test;

import java.util.Arrays;
import java.util.LinkedList;
import java.util.List;
import java.util.Queue;

public class TreeParser {

    public static TreeNode parseInput(Integer... a) {
        return parseInput(Arrays.asList(a));
    }
    public static TreeNode parseInput(List<Integer> input) {
        //return parseInput(input, 0);
        var root = new TreeNode(input.get(0));
        var index = 1;
        Queue<TreeNode> nodesToProcess = new LinkedList<>();
        nodesToProcess.add(root);
        while(index < input.size()) {
            var currentNode = nodesToProcess.poll();
            Integer valLeft = input.get(index++);
            Integer valRight = input.get(index++);
            if (valLeft != null) {
                var left = new TreeNode(valLeft);
                left.parent = currentNode;
                currentNode.left = left;
                nodesToProcess.add(left);
            }
            if (valRight != null) {
                var right = new TreeNode(valRight);
                right.parent = currentNode;
                currentNode.right = right;
                nodesToProcess.add(right);
            }
        }
        return root;

    }
}

File: python/test/model/tree_parser.py
from collections import deque
from typing import List

from python.test.model.tree_node import TreeNode


def parse_tree(data: List[int]) -> TreeNode:
    root = TreeNode(data[0])
    index = 1
    nodes_to_process = deque([root])
    while nodes_to_process:
        current_node = nodes_to_process.popleft()
        if index < len(data):
            val_left = data[index]
            index += 1
            if val_left is not None:
                left = TreeNode(val_left)
                left.parent = current_node
                current_node.left = left
                nodes_to_process.append(left)
        if index < len(data):
            val_right = data[index]
            index += 1

            if val_right is not None:
                right = TreeNode(val_right)
                right.parent = current_node
                current_node.right = right
                nodes_to_process.append(right)

    return root
